src/pyfaf/actions/pull_associates.py

```diff
@@ -33,66 +33,42 @@ def __init__(self):
 
     def run(self, cmdline, db):
         if len(cmdline.opsys) == 0:
-            tasks = []
-            for opsys in systems.values():
-                releases = get_releases(db, opsys_name=opsys.nice_name)
-                tasks += [(opsys, release) for release in releases if
-                          release.status != "EOL"]
-        elif len(cmdline.opsys) == 1:
-            shortname = cmdline.opsys[0]
+            cmdline.opsys = systems.keys()
+
+        opsyss = []
+        for shortname in cmdline.opsys:
             if shortname not in systems:
-                self.log_error("Operating system '{0}' is not installed"
-                               .format(shortname))
-                return 1
+                self.log_warn("Operating system '{0}' is not installed"
+                              .format(shortname))
```

Comment:
while in this, can you change it to lazy evaluation? `log_warn("foo {0}", shortname)`

Reply:
You mean `log_warn("foo %s", shortname)`? If so, I can change it, but it is `the old style` and I am not sure, if the laziness there makes that much of a difference.